bigquery/google/cloud/bigquery/client.py

```diff
@@ -942,6 +942,25 @@ def list_rows(self, table, selected_fields=None, max_results=None,
         iterator.schema = schema
         return iterator
 
+    def list_partitions(self, table):
+        """List the partitions in a table.
+
+        :type table: One of:
+                     :class:`~google.cloud.bigquery.table.Table`
+                     :class:`~google.cloud.bigquery.table.TableReference`
+        :param table: the table to list, or a reference to it.
+
+        :rtype: list
+        :returns: a list of time partitions
+        """
+        config = QueryJobConfig()
+        config.use_legacy_sql = True  # required for '$' syntax
+        rows = self.query_rows(
+            'SELECT partition_id from [%s.%s$__PARTITIONS_SUMMARY__]' %
```

Comment:
We should probably add the project ID, too, since they might want to list partitions on any table reference, not just one with this client's project.

`'SELECT partition_id from [%s:%s.%s$__PARTITIONS_SUMMARY__]' % (table.project, table.dataset_id, table.table_id)`

Reply:
Done.